### PR TITLE
atsamd2x_xpro: follow best practices.

### PR DESCRIPTION
**Pull request by Flameeyes**
atsamd2x_xpro: follow best practices.

This updates the configurations for these two boars to match the expected
best practices for new boards, as discussed in #61140.

Signed-off-by: Diego Elio Pettenò <flameeyes@meta.com>

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/zephyrproject-rtos/zephyr/pull/61437).
* #61529
* __->__ #61437

### CODE REVIEW

**Comment by nandojve** [comment]
Hi @Flameeyes ,

The below list tells to CI that it needs build those matches for this board. This means that any wrong/missing configuration on the samples/tests should be fixed. See CI errors for more details.

```diff
+supported:
+  - adc
+  - gpio
+  - i2c
+  - spi
+  - watchdog
```

**Comment by Flameeyes** [comment]
Yep will be looking at them tonight, I couldn't find an easy way to consistently rerun all the tests locally.

The good thing is that I *do* have the SAMD20 XPRO on the desk so I can make sure it still works right 😅

File: boards/arm/atsamd20_xpro/doc/index.rst [hunk 1]
```diff
@@ -48,6 +51,8 @@ features:
 +-----------+------------+------------------------------------------+
 | USART     | on-chip    | Serial ports                             |
 +-----------+------------+------------------------------------------+
+| I2C       | on-chip    | I2C ports                                |
++-----------+------------+------------------------------------------+
```
Comment (nandojve) [inline boards/arm/atsamd20_xpro/doc/index.rst:55]: ADC still missing.

**Review by nandojve — CHANGES_REQUESTED** [review]
Overall LGTM!

File: boards/arm/atsamd20_xpro/doc/index.rst (hunk 1 above)
Comment (Flameeyes) [reply boards/arm/atsamd20_xpro/doc/index.rst:57]: Updated.

**Review by nandojve — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
LGTM

**Pull request merged**